admin/includes/classes/AdminRequestSanitizer.php

```diff
@@ -535,22 +569,98 @@ private function filterMultiDimensional($parameterName, $parameterDefinition)
                 $this->runSpecificSanitizer($parameterName, $newParameterDefinition);
                 $newKey = $_POST[$parameterName];
                 $requestPost[$parameterName][$newKey] = $hacked;
+            } else if ($this->doStrictSanitization) {
+                unset($requestPost[$parameterName][$key]);
+                unset($_POST);
+                $_POST[$parameterName] = $key;
+                $this->arrayName = $newArrayName;
+                $this->filterStrictSanitizeKeys();
+                if (array_key_exists($pkey, $_POST)) {
+                    $this->filterStrictSanitizeValues();
+                    $newKey = $_POST[$parameterName];
+                    $this->arrayName = $newArrayName;
+                    $requestPost[$parameterName][$newKey] = $hacked;
+                } else {
+                    continue; // Key is "unclean" and therefore should use the next key.
+                }
             }
+            $newArrayName = $newArrayName . '[' . $newKey . ']';
```

Comment:
Arghh.. Another version of the assignment with this being a manual version that did not fit the previous pattern for replacement which means more revision in this area to make sure it works as expected/desired when/if replaced with `$newArrayName = $this->setCurrentArrayName($newKey)`.